### DISCUSSION TITLE
How to send a break signal to serial device?

### DISCUSSION THREAD

**Discussion by XeCycle** [wezterm/wezterm#4751] [Q&A]
### What Operating System(s) are you running on?

Windows

### WezTerm version

20230712-072601-f4abf8fd

### Ask your question!

I use wezterm on windows mainly for connecting to serial devices.  Some of my devices support baudrate auto-configuration; I can pick an arbitrary baudrate when connecting, and send a break signal, then the device will follow this baudrate.  It's useful when device boots with a default of 9600, and I can easily connect to it at 115200.  GNU screen uses ^A b for this; how can I do it in wezterm?

**Comment by wez** [comment]
hmm, there isn't an explicit API for this in the serial port handling crates used by wezterm.
You might try assigning a key to [SendString](https://wezfurlong.org/wezterm/config/lua/keyassignment/SendString.html) and sending a NUL byte in the hope that that is interpreted as a break signal somewhere along the way, but I don't know if that behavior is guaranteed:

`wezterm.action.SendString '\x00'`
